org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/rename/RenameService2.java

```diff
@@ -163,6 +150,24 @@ public WorkspaceEdit rename(IRenameService2.Options options) {
 			throw Exceptions.sneakyThrow(e);
 		}
 	}
+	
+	protected void applyModifications(EObject element, WorkspaceEdit workspaceEdit, ServerRefactoringIssueAcceptor issueAcceptor, IRenameService2.Options options) {
```

Comment:
I know this does already have an aweful signature, but the languageserver.context could also be useful to implement a reasonable rename strategy. Also this method deserves some JavaDoc that explains, that the element is from a live-scope-RS and this entire thing runs in a read-transaction on the index